### PR TITLE
Improve release process

### PR DESCRIPTION
**Pull request by pwittrock**
- Get rid of releaseall.sh -- it is too error prone right now
- Support for releasing pluginator
- Automatically release binaries for kustomize
- Update instructions in README.md
- Move old instructions to legacyRelease.md

### CODE REVIEW

File: releasing/VERSIONS [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # kyaml version
 export kyaml_major=0
 export kyaml_minor=0
-export kyaml_patch=1
+export kyaml_patch=2
```
Comment (Liujingfang1) [inline releasing/VERSIONS]: Are the version in this file for the next release versions? Should it be reviewed first before cutting the release?

File: releasing/README.md [hunk 2]
```diff
@@ -110,196 +110,69 @@ infrequently).
...
-```
-git checkout -b $branch
-```
-
-### define the release tag
+For modules with a major version of 1 or greater:
+ 
+- if no new features are added in the release and it is only bug fixes, update the patch version
+  - `kustomize_patch=0` = `kustomize_patch=1`
+- if new features are added in the release, update the minor version
```
Comment (Liujingfang1) [inline releasing/README.md]: And change the patch version to 0

File: releasing/README.md [hunk 3]
```diff
@@ -110,196 +110,69 @@ infrequently).
...
+  - `kustomize_minor=5` = `kustomize_minor=6`
+- if the API surface is changed in a non-backward compatible way, update the major version
```
Comment (Liujingfang1) [inline releasing/README.md]: Change the minor version and patch version to 0.

File: releasing/VERSIONS (hunk 1 above)
Comment (pwittrock) [reply releasing/VERSIONS]: the tooling respects either.  up to us what we want our process to be.

File: releasing/README.md (hunk 3 above)
Comment (pwittrock) [reply releasing/README.md]: done

File: releasing/README.md (hunk 2 above)
Comment (pwittrock) [reply releasing/README.md]: done

File: releasing/README.md [hunk 4]
```diff
@@ -110,196 +110,72 @@ infrequently).
...
+ 
+- if no new features are added in the release and it is only bug fixes
+  - update the patch version: `kustomize_patch=0` => `kustomize_patch=1`
+- if new features are added in the release
+  - update the minor version: `kustomize_minor=5` => `kustomize_minor=6`
+  - set the patch version to 0 
+- if the API surface is changed in a non-backward compatible way
+  - update the major version `kustomize_major=3` => `kustomize_major=4`
+  - set the minor version to 0
+  - set the patch version to 0
```
Comment (monopole) [inline releasing/README.md]: FWIW, all i want is
```
release (kustomize|api|kyaml|...) (major|minor|patch)
```
per the TODO being deleted.

The script doing this - which i think should be a go program for sanity - can get the right digits by consulting existing git tags.

There's enough information in the tags to get this done without error.

BUT am willing to live with this as a stopgap

File: releasing/README.md [hunk 5]
```diff
@@ -110,196 +110,72 @@ infrequently).
...
-```
-git tag -l | grep $module
-```
-
-Remote:
-```
-git ls-remote --tags upstream | grep $module
-```
-
-### determine the version
```
Comment (monopole) [inline releasing/README.md]: i think this little table was handy to have

**Comment by pwittrock** [comment]
I've dropped the documentation changes.  Now this only fixes some issues with the scripts.

**Pull request merged**